packages/salesforcedx-sobjects-faux-generator/src/generator/fauxClassGenerator.ts

```diff
@@ -221,6 +239,82 @@ export class FauxClassGenerator {
     return this.successExit();
   }
 
+  public async generateMin(
```

Comment:
I created this reduced method since we'll keep iterative over this approach as we get more feedback and didn't want to make `generate` harder to maintain.

Reply:
Makes sense. Although it seems like it'd be a relatively small change in the current generate to support this - is cloning the implementation any easier to maintain/less confusing?

Reply:
I think we'll keep iterating over this and I want to be able to do it fast enough without breaking the other commands that use the `generate` piece. If it turns out that the changes are not very specific then I'll definitely merge `generate` and `generateMin`.

Reply:
@lcampos @brpowell Not specific to this PR - What do you think if we parallelize batch fetching? For example in my example org there are 461 standard sobjects, and batch comes in size of 25. There are 19 requests made sequentially.

Here's a piece of sample code in `generate`:
```
const filteredSObjects = sobjects.filter(this.isRequiredSObject);
    let j = 0;
    const requests = [];
    while (j < filteredSObjects.length) {
      requests.push(describe.describeSObjectBatch(filteredSObjects, j));
      j += 25;
    }

    try {
      if (
        this.cancellationToken &&
        this.cancellationToken.isCancellationRequested
      ) {
        return this.cancelExit();
      }
      const results = await Promise.all(requests);
      fetchedSObjects = ([] as SObject[]).concat(...results);
    } catch (errorMessage) {
      return this.errorExit(
        nls.localize('failure_in_sobject_describe_text', errorMessage)
      );
    }

    // tslint:disable-next-line:prefer-for-of
    for (let i = 0; i < fetchedSObjects.length; i++) {
      if (fetchedSObjects[i].custom) {
        customSObjects.push(fetchedSObjects[i]);
      } else {
        standardSObjects.push(fetchedSObjects[i]);
      }
    }
```

Measurement for parallelization tested on refreshing standard sobjects (3 runs): 
sequential: 26864.92 ms
parallelized: 3584.46 ms

Reply:
We should definitely look into changing this in another PR - I don't see any reason why this can't be parallelized

Reply:
@brpowell There's concurrent API request limit enforced on requests with 20s or longer, which would hardly be a case here, but we should have some error handling around that. https://developer.salesforce.com/docs/atlas.en-us.salesforce_app_limits_cheatsheet.meta/salesforce_app_limits_cheatsheet/salesforce_app_limits_platform_api.htm

Discussed with @lcampos offline - we could make this enhancement on PWT's work on `@salesforce/core` which will replace this code